components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -3314,8 +3314,14 @@ public static Certificate getX509CertOfOAuthApp(String clientId,
                 // Build the Certificate object from cert content.
                 return IdentityUtil.convertPEMEncodedContentToCertificate(certificateContent);
             } else {
-                throw new IdentityOAuth2Exception("Public certificate not configured for Service Provider with " +
-                        "client_id: " + clientId + " of tenantDomain: " + tenantDomain);
+                String errorMsg = StringUtils.EMPTY;
+                if (checkIfIDTokenEncryptionEnabled(clientId)) {
```

Comment:
Shouldn't this be done at the place where we do encryption?

Reply:
Addressed with 63311097ce87f4c33248d9ac0dc9e7544403f4cd